### PR TITLE
Features/ncne aduser changes

### PR DESCRIPTION
**Pull request by ssrajan**
The following changes are done
1. Database and EF changes for AdUser
2. Landing page changes and change of Cog to buttons to Assign user and notes
3. New page , Validation and type-ahead changes
4. Workflow page, validation and type-ahead for roles
5. comments and Stage comments
6. Create CARIS project
7. Historical Task display and search

The following are commented , will be done in the next PR
1. Database integrity Tests
2. Comments helper Tests
3. Historical Tasks Tests
4. new Tasks Tests
5. Page Validation Tests
6. Workflow Tests

### CODE REVIEW

File: src/DbUpdatePortal/Pages/Index.cshtml [hunk 1]
```diff
@@ -123,9 +123,9 @@
             <tbody>
 
                 @foreach (var task in Model.DbUpdateTasks.Where(t => t.Status == DbUpdateTaskStatus.InProgress.ToString()
-                                                          &&
-                                                          (t.TaskRole.Verifier.DisplayName == Model.CurrentUser.DisplayName
-                   || t.TaskRole.Compiler.DisplayName == Model.CurrentUser.DisplayName)))
+                                                         &&
+                                                         (t.TaskRole.Verifier.DisplayName == Model.CurrentUser.DisplayName
+                  || t.TaskRole.Compiler.DisplayName == Model.CurrentUser.DisplayName)))
```
Comment (benbhall) [inline src/DbUpdatePortal/Pages/Index.cshtml:128]: Can we make this comparison on the UserPrincipalName?

File: src/NCNEPortal/Data/HpdUsers.json [hunk 2]
```diff
@@ -1,42 +1,19 @@
 [
   {
-    "AdUsername": "Greg Williams",
+    "AdUserId": 3,
```
Comment (benbhall) [inline src/NCNEPortal/Data/HpdUsers.json:3]: Caris users in HPD are case sensitive so that will need to be TM_USER.

Comment (benbhall) [reply src/NCNEPortal/Data/HpdUsers.json:3]: Bah, that's highlighted completely wrong piece of code. I was referring to the `"TM_User"` bit.

File: src/Common.Helpers/DatabasesHelpers.cs [hunk 3]
```diff
@@ -118,7 +118,7 @@ public static void ClearNcneWorkflowDbTables(NcneWorkflowDbContext dbContext, bo
             dbContext.WorkflowType.RemoveRange(dbContext.WorkflowType);
 
             dbContext.HpdUser.RemoveRange(dbContext.HpdUser);
-            dbContext.AdUser.RemoveRange(dbContext.AdUser);
+            //dbContext.AdUser.RemoveRange(dbContext.AdUser);
```
[the remark this answers is not in the document]
Comment (ssrajan) [reply src/Common.Helpers/DatabasesHelpers.cs]: removed the comments

File: src/NCNEPortal/Data/HpdUsers.json (hunk 2 above)
Comment (ssrajan) [reply src/NCNEPortal/Data/HpdUsers.json:3]: changed the TM_User

File: src/DbUpdatePortal/Pages/Index.cshtml (hunk 1 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Pages/Index.cshtml:128]: changed the condition to userPrincipalname

**Pull request merged**